docs/content/doc/installation/database-preparation.en-us.md

```diff
@@ -154,3 +154,136 @@ Note: All steps below requires that the database engine of your choice is instal
     where `gitea` is database user, `giteadb` is database name, and `203.0.113.3` is IP address of your database instance.
 
     You should be prompted to enter password for the database user, and connected to the database.
+
+## Database Connection over TLS
+
+If you have database instance on private network, this section can be omitted, since the security between Gitea and database instance is guaranteed. If instead the database instance is on public network, use TLS to encrypt connection to database, as it is possible for third-party to intercept database connection.
+
+### Prerequisites
+
+- You need two valid TLS certificate, one for database instance (database server) and one for Gitea instance (database client). Both certificates must be signed by trusted CA.
+- `X509v3 Extended Key Usage` extension attribute must have `TLS Web Server Authentication` for database server certificate, and `TLS Web Client Authentication` for database client certificate.
+- On database server certificate, `Subject Alternative Name` or `Common Name` entry must be fully-qualified domain name (FQDN) for database instance. On database client certificate, either entry above must be database username that Gitea instance will be used to connect.
+- You need mapping from domain name of both Gitea and database instances to their respective IP addresses. Either set up DNS records or add local mapping to `/etc/hosts`. This allows database connections to be performed by domain name instead of IP address. See documentation of your system for details.
```

Comment:
```suggestion
- You need domain name mappings of both Gitea and database servers to their respective IP addresses. Either set up DNS records for them or add local mappings to `/etc/hosts` (`%WINDIR%\System32\drivers\etc\hosts` in Windows) on each system. This allows the database connections to be performed by domain name instead of IP address. See documentation of your system for details.
```